### PR TITLE
ACME plugin improvement

### PR DESCRIPTION
**Pull request by hosseinsh**
Lemur's ACME DNS plugin was not considering already valid hostnames, and repeating the expensive DNS domain ownership validation. This improvement would allow the ACME DNS to benefit from still valid hostnames.

### CODE REVIEW

File: lemur/plugins/lemur_acme/acme_handlers.py [hunk 1]
```diff
@@ -336,10 +342,14 @@ def start_dns_challenge(
 
         change_ids = []
         cname_delegation = domain != target_domain
-        dns_challenges = self.get_dns_challenges(domain, order.authorizations)
+        # This method will consider and skip valid HTTP01 challenges
+        dns_challenges, hostname_still_validatd = self.get_dns_challenges(domain, order.authorizations)
```
Comment (charhate) [inline lemur/plugins/lemur_acme/acme_handlers.py:346]: `hostname_still_validatd ` --> `hostname_still_validated`

**Comment by hosseinsh** [comment]
need to update the tests

File: lemur/plugins/lemur_acme/tests/test_acme_dns.py [hunk 2]
```diff
@@ -49,8 +51,28 @@ def test_get_dns_challenges(self, mock_len):
...
+    @patch("lemur.plugins.lemur_acme.plugin.len", return_value=1)
+    def test_get_dns_challenges_already_valid(self, mock_len):
+        assert mock_len
+
+        from acme import challenges
+
+        host = "example.com"
+        c = challenges.DNS01()
+
+        mock_authz = MagicMock
```
Comment (charhate) [inline lemur/plugins/lemur_acme/tests/test_acme_dns.py:67]: `MagicMock` --> `MagicMock()`

**Review by charhate — APPROVED** [review]

File: lemur/plugins/lemur_acme/tests/test_acme_dns.py (hunk 2 above)
Comment (hosseinsh) [reply lemur/plugins/lemur_acme/tests/test_acme_dns.py]: upps, thanks for catching that.

**Pull request merged**